tests/debugging/exception/test_auto_instrument.py

```diff
@@ -0,0 +1,75 @@
+from contextlib import contextmanager
+
+import pytest
+
+import ddtrace
+from tests.debugging.mocking import debugger
+from tests.utils import TracerTestCase
+
+
+@contextmanager
+def auto_instrument():
+    with debugger(exception_debugging=True) as d:
+        yield d
+
+
+class ExceptionDebuggingTestCase(TracerTestCase):
+    def setUp(self):
+        super(ExceptionDebuggingTestCase, self).setUp()
+        self.backup_tracer = ddtrace.tracer
+        ddtrace.tracer = self.tracer
+
+    def tearDown(self):
+        ddtrace.tracer = self.backup_tracer
+        super(ExceptionDebuggingTestCase, self).tearDown()
+
+    def test_debugger_exception_debugging(self):
+        def a(v, d=None):
+            with self.trace("a"):
+                if not v:
+                    raise ValueError("hello", v)
+
+        def b(bar):
+            with self.trace("b"):
+                m = 4
+                a(bar % m)
+
+        def c(foo=42):
+            with self.trace("c"):
+                sh = 3
+                b(foo << sh)
+
+        with auto_instrument() as d:
+            with pytest.raises(ValueError):
+                c()
+
+            self.assert_span_count(3)
+            assert len(d.test_queue) == 3
+
+            snapshots = {str(s.uuid): s for s in d.test_queue}
+
+            for n, span in enumerate(self.spans):
+                assert span.get_tag("error.debug-info-captured") == "true"
+
+                exc_id = span.get_tag("_dd.debug.error.exception-id")
+
+                info = {k: v for k, v in enumerate(["c", "b", "a"][n:], start=1)}
+
+                for i in range(1, len(info) + 1):
+                    fn = info[i]
+
+                    # Check that we have all the tags for each snapshot
+                    assert span.get_tag("_dd.debug.error.%d.snapshot.id" % i) in snapshots
```

Comment:
should we also insure the right frame was captured by the snapshot? maybe check the snapshot stack or local values?